### DISCUSSION TITLE
Roadmap for when `bobgen-sql` might support mysql dialect

### DISCUSSION THREAD

**Discussion by warbaque** [stephenafamo/bob#282] [Q&A] [answered]
Is there currently any roadmap or estimate when `bobgen-sql` might [support mysql dialect](https://github.com/stephenafamo/bob/blob/26207d046773fb4dba64b282c5412031b4deff45/gen/bobgen-sql/main.go#L77)?

If it's just few months I think we can do with our janky solution in the mean time. But if there's no plans for it at the moment, we need a more robust setup.

Currently generating models from (mysql) database schema requires:
- starting throwaway db
- running all migrations
- generating code with `bobgen-mysql`
- stopping db

It would be nice if I could just pass db/migration folder to generator without starting the actual db. Is there any estimate when this might be possible?

**Comment by stephenafamo** [comment, marked as answer]
It depends on when it would be possible to find a way to run MySQL in-process.

Previously, I had tried with <https://github.com/dolthub/go-mysql-server>, but it was blocked by several issues, such as [this one](https://github.com/dolthub/dolt/issues/3168).

Now that that issue has recently been fixed, I'll try again and see if I can make it work.

**Reply by warbaque** [reply]
Great to hear. Currently I run all steps with `docker compose -f dev/autogen.yml run --rm bobgen-mysql`, but it would be nice if I could also just run locally `go generate ./db`

**Reply by stephenafamo** [reply]
@warbaque I found some time to work on this and it _almost_ works. See #292

You can try it out in the `mysql-from-file` branch. However, tests currently do not pass because:

1. The output of the golden file does not completely match what a regular server produces
2. The migration fails when trying to create indexes from expressions `CREATE INDEX idx1 ON test_index_expressions ((col1 + col2));`. Relevant issue is [#2945](https://github.com/dolthub/dolt/issues/2945)

**Reply by stephenafamo** [reply]
Now supported in `v0.35.0` by using [The Testcontainers module for MySQL ](https://golang.testcontainers.org/modules/mysql)
